cognite/neat/rules/models/_types.py

```diff
@@ -82,15 +90,41 @@ def _validator(value: Any, next_: Any, ctx: ValidationInfo) -> Any:
 ]
 
 
-def _property_validation(value: str) -> str:
-    if not PATTERNS.property_id_compliance.match(value):
-        raise RegexViolationError(value, PROPERTY_ID_COMPLIANCE_REGEX)
+def _property_validation(value: str, property_type: EntityTypes) -> str:
+    compiled_regex = PATTERNS.entity_pattern(property_type)
+    if not compiled_regex.match(value):
+        raise RegexViolationError(value, compiled_regex.pattern)
```

Comment:
Good case for a decorator??
```suggestion
def _property_validation_factory(property_type: EntityTypes):
    def _property_validation(value: str) -> str:
        compiled_regex = PATTERNS.entity_pattern(property_type)
        if not compiled_regex.match(value):
            raise RegexViolationError(value, compiled_regex.pattern)
        if PATTERNS.more_than_one_alphanumeric.search(value):
            warnings.warn(
                RegexViolationWarning(
                    value,
                    compiled_regex.pattern,
                    "property",
                    "MoreThanOneNonAlphanumeric",
                ),
                stacklevel=2,
            )
        return value
    return _property_validation
```